core/server/master/src/main/java/alluxio/master/file/cmdmanager/FileSystemCommandManager.java

```diff
@@ -0,0 +1,371 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.file.cmdmanager;
+
+import static alluxio.master.file.cmdmanager.command.CommandInfo.CmdOptions;
+
+import alluxio.exception.AlluxioRuntimeException;
+import alluxio.exception.status.ResourceExhaustedException;
+import alluxio.job.meta.JobIdGenerator;
+import alluxio.job.wire.Status;
+import alluxio.master.file.cmdmanager.command.CmdType;
+import alluxio.master.file.cmdmanager.command.CommandInfo;
+import alluxio.master.file.cmdmanager.command.ExecutionStatus;
+import alluxio.master.file.cmdmanager.command.Task;
+import alluxio.master.journal.DelegatingJournaled;
+import alluxio.master.journal.Journaled;
+import alluxio.master.journal.checkpoint.CheckpointName;
+import alluxio.proto.journal.File;
+import alluxio.proto.journal.Journal;
+import alluxio.resource.CloseableIterator;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * The Filesystem Command manager which controls distributed job operations.
+ */
+public final class FileSystemCommandManager implements DelegatingJournaled {
```

Comment:
`DelegatingJournaled` is convenient if you are delegating to an existing journal implementation. But since we are implementing a new one anyways, just have this implement `Journaled` and you won't need the `CmdEntry` class.